src/AppBundle/Service/TeamMembershipService.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace AppBundle\Service;
+
+use AppBundle\Event\TeamMembershipEvent;
+use Doctrine\ORM\EntityManager;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class TeamMembershipService
+{
+    private $em;
+    private $dispatcher;
+
+    public function __construct(EntityManager $em, EventDispatcherInterface $dispatcher)
+    {
+        $this->em = $em;
+        $this->dispatcher = $dispatcher;
+    }
+
+    public function updateTeamMemberships()
+    {
+        $teamMemberships = $this->em->getRepository('AppBundle:TeamMembership')->findActiveTeamMemberships();
+        foreach ($teamMemberships as $teamMembership) {
+            $endSemester = $teamMembership->getEndSemester();
+            if ($endSemester) {
+                if ($endSemester->getSemesterEndDate() <= $this->em->getRepository('AppBundle:Semester')->findCurrentSemester()->getSemesterStartDate()) {
```

Comment:
Vil dette noen gang bli true, siden kun aktive teamMemberships hentes ut: `$teamMemberships = $this->em->getRepository('AppBundle:TeamMembership')->findActiveTeamMemberships();`?